apps/challenges/views.py

```diff
@@ -424,6 +424,27 @@ def get_challenge_by_pk(request, pk):
         return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
 
 
+@api_view(["GET"])
+@throttle_classes([AnonRateThrottle])
+def get_challenge_by_title(request, title):
+    """
+    Returns a particular challenge by title. The string in request should exactly match the title of challenge.
+    """
+    challenges = Challenge.objects.filter(
+        title__icontains=title, approved_by_admin=True, published=True
+    )
+    response_data = []
+    for challenge in challenges:
+        serializer = ChallengeSerializer(
+            challenge, context={"request": request}
+        )
+        response_data.append(serializer.data)
+    if len(response_data) == 0:
+        exception_data = {"error": "Challenge does not exist!"}
```

Comment:
{"error": "Challenge with the title {} does not exist!".format(challenge_title)}

Reply:
Updated the error message in f624c2e